utils/sampler/lib/detail_view.dart

```diff
@@ -0,0 +1,156 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:file/file.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:path/path.dart' as path;
+import 'package:snippets/snippets.dart';
+
+import 'helper_widgets.dart';
+import 'model.dart';
+
+/// The detail view that shows the details of a particular sample.
+///
+/// And
+class DetailView extends StatefulWidget {
+  const DetailView({Key? key}) : super(key: key);
+
+  @override
+  _DetailViewState createState() => _DetailViewState();
+}
+
+// Used to run export in another isolate.
+Future<void> _doExport(FlutterSampleLiberator project) async {
+  await project.extract(overwrite: true);
+}
+
+class _DetailViewState extends State<DetailView> {
+  FlutterSampleLiberator? project;
+  bool exporting = false;
+  bool importing = false;
+
+  void _extractSample() {
+    setState(() {
+      exporting = true;
+      if (project == null) {
+        final Directory outputLocation =
+            Model.instance.filesystem.systemTempDirectory.createTempSync('flutter_sample.');
+        project = FlutterSampleLiberator(
+          Model.instance.currentElement!,
+          Model.instance.currentSample!,
+          location: outputLocation,
+          flutterRoot: Model.instance.flutterRoot,
+        );
+      }
+      compute(_doExport, project!).whenComplete(() {
+        setState(() {
+          exporting = false;
+        });
+      });
+    });
+  }
+
+  void _reinsertIntoFrameworkFile(BuildContext context) {
+    setState(() {
+      if (project == null) {
+        return;
+      }
+      importing = true;
+      project!.reinsert().then((String error) {
+        if (error.isEmpty) {
+          return;
+        }
+        final ScaffoldMessengerState? scaffold = ScaffoldMessenger.maybeOf(context);
+        scaffold?.showSnackBar(
+          SnackBar(
+            content: Text(error),
+          ),
+        );
+      }).whenComplete(() {
+        setState(() {
+          importing = false;
+        });
+      });
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    if (Model.instance.currentSample == null) {
+      return const Scaffold(body: Center(child: Text('Working sample not set.')));
+    }
+    final CodeSample sample = Model.instance.currentSample!;
+    final String filename = sample.start.file != null
+        ? path.relative(sample.start.file!.path, from: Model.instance.flutterRoot.path)
+        : '<generated>';
+    return Scaffold(
+      appBar: AppBar(
+        title: Text('${sample.element} - $filename:${sample.start.line}'),
+        actions: const <Widget>[],
```

Comment:
Unnecessary? Or does this remove some default actions or something?

Reply:
No, just unnecessary. Removed.